### PR TITLE
* gdal/PROVENANCE.TXT: Update to remove directories that no longer exist.

### PR DESCRIPTION
**Pull request by bje-**
The history, while interesting, can be preserved in version control. I think it's better that the content of the file corresponds to the source tree.

### CODE REVIEW

File: gdal/PROVENANCE.TXT [hunk 1]
```diff
@@ -139,56 +139,21 @@ Note: all the following are build options, not required.
...
-* gdal_wrap.c: Generated with SWIG under very permissive license (noted in file).
-* Scripts mostly by me, all ok.
-
-=== gdal/pymod/samples ===
-
-* gdal_vrtmerge.py: Under LGPL (derived from gdal_merge.py), several contributors.
-* histrep.py: corrected copyright message (Atlantis to FrankW), change license to MIT/X.
-* various contributors, all under proper and compatible copyright headers.
-
-=== gdal/ogr/ogrsf_rmts/generic ===
```
Comment (rouault) [inline gdal/PROVENANCE.TXT]: This is still valid. There's just a typo in the directory name. should read gdal/ogr/ogrsf_frmts/generic

File: gdal/PROVENANCE.TXT [hunk 2]
```diff
@@ -139,56 +139,21 @@ Note: all the following are build options, not required.
...
-
-* Some files here (and elsewhere in OGR) copyright Softmap Inc (but MIT/X).
-
 === gdal/ogr/ogrsf_frmts/avc ===
 
 * Some copyright Daniel Morissette, MIT/X.
 * Included copy of dbfopen.h from Shapelib.  We really ought to reference the one in ../shape.
 
-=== gdal/ogr/ogrsf_rmts/csv ===
```
Comment (rouault) [inline gdal/PROVENANCE.TXT]: still valid. just typo on directory name

File: gdal/PROVENANCE.TXT [hunk 3]
```diff
@@ -139,56 +139,21 @@ Note: all the following are build options, not required.
...
-
-* drv_*.html not copyright.
-
-=== gdal/ogr/ogrsf_rmts/dgn ===
```
Comment (rouault) [inline gdal/PROVENANCE.TXT]: still valid. just typo on directory name

File: gdal/PROVENANCE.TXT [hunk 4]
```diff
@@ -429,12 +369,6 @@ I have added this license to LICENSE.TXT to satisfy the credit requirement.
 
 * FrankW, clean.
 
-=== gdal/frmts/hdf ===
```
Comment (rouault) [inline gdal/PROVENANCE.TXT]: I think this should read gdal/frmts/hfa

File: gdal/PROVENANCE.TXT [hunk 5]
```diff
@@ -429,12 +369,6 @@ I have added this license to LICENSE.TXT to satisfy the credit requirement.
...
-
-* geoextra.c: Derived by public domain contribution to libgeotiff from PCI but the terms are not clear.  This is a libgeotiff issue.  There is no problem with the license - it just isn't well documented in the code file.
-* Mostly copyright Intergraph Corporation, clean.
```
Comment (rouault) [inline gdal/PROVENANCE.TXT]: This line and the following lines are still valid. Only the one about geoextra.c is no longer current

File: gdal/PROVENANCE.TXT (hunk 1 above)
Comment (bje-) [reply gdal/PROVENANCE.TXT]: Thanks. I found all of the missing directories with an `awk` script, so didn't check manually.

**Pull request merged**